src/datastore/postgres-store.ts

```diff
@@ -3108,6 +3108,53 @@ export class PgDataStore
     return tx;
   }
 
+  async getMempoolTxs(args: {
+    txIds: string[];
+    includeUnanchored: boolean;
+    includePruned?: boolean;
+  }) {
+    return this.queryTx(async client => {
+      const hexTxIds = args.txIds.map(txId => hexToBuffer(txId));
+      const result = await client.query<MempoolTxQueryResult>(
+        `
+        SELECT ${MEMPOOL_TX_COLUMNS}
+        FROM mempool_txs
+        WHERE tx_id = ANY($1)
+        `,
+        [hexTxIds]
+      );
+      if (result.rowCount === 0) {
+        return [];
+      }
+      const notPruned = result.rows.filter(memTx => memTx.pruned && !args.includeUnanchored);
+      if (notPruned.length !== 0) {
+        const unanchoredBlockHeight = await this.getMaxBlockHeight(client, {
+          includeUnanchored: true,
+        });
+        const notPrunedBufferTxIds = notPruned.map(tx => tx.tx_id);
+        const query = await client.query<{ tx_id: Buffer }>(
+          `
+            SELECT tx_id
+            FROM txs
+            WHERE canonical = true AND microblock_canonical = true 
+            AND tx_id = ANY($1)
+            AND block_height = $2
+            `,
+          [notPrunedBufferTxIds, unanchoredBlockHeight]
+        );
+        // The tx is marked as pruned because it's in an unanchored microblock
+        query.rows.forEach(tran => {
+          const transaction = result.rows.find(tx => tx.tx_id === tran.tx_id);
+          if (transaction) {
+            transaction.pruned = false;
+            transaction.status = DbTxStatus.Pending;
+          }
+        });
+      }
+      return result.rows.map(transaction => this.parseMempoolTxQueryResult(transaction));
```

Comment:
There's non-trivial logic duplicated in this function (originally in `getMempoolTx`), is it possible to share the logic between the two?